datafusion/physical-plan/src/aggregates/group_values/group_column.rs

```diff
@@ -62,62 +62,96 @@ pub trait GroupColumn: Send + Sync {
     fn take_n(&mut self, n: usize) -> ArrayRef;
 }
 
+/// Stores a collection of primitive group values which are known to have no nulls
+#[derive(Debug)]
+pub struct NonNullPrimitiveGroupValueBuilder<T: ArrowPrimitiveType> {
```

Comment:
I think NonNull can be a (boolean) generic parameter as well to avoid code duplication...

Reply:
I will look into it

Reply:
added as a subtask of https://github.com/apache/datafusion/issues/12680

Reply:
https://github.com/apache/datafusion/pull/12703